src/sparseml/openpifpaf/train.py

```diff
@@ -151,11 +152,37 @@ def main():
 
     datamodule = openpifpaf.datasets.factory(args.dataset)
 
+    """
+    Cases for checkpointing:
+    1. No checkpoint - no changes to base flow
+    2. Checkpoint from local file - no changes to base flow
+    3. Checkpoint from zoo stub:
+        a. set args.checkpoint to None to get a base net
+        b. manually load in state dict
+    """
+    zoo_stub = None
+    if args.checkpoint and args.checkpoint.startswith("zoo:"):
+        if args.basenet is None:
+            raise ValueError(
+                "Must specify --basenet when using a zoo stub in --checkpoint"
+            )
+        zoo_stub = args.checkpoint
+        args.checkpoint = None
+
+    # NOTE: configure this after removing zoo stub checkpoint
+    openpifpaf.network.Factory.configure(args)
+
     net_cpu, start_epoch = openpifpaf.network.Factory().factory(
         head_metas=datamodule.head_metas
     )
     loss = openpifpaf.network.losses.Factory().factory(datamodule.head_metas)
 
+    if zoo_stub is not None:
+        # just load state dict from zoo stub
+        checkpoint_path = download_framework_model_by_recipe_type(Model(zoo_stub))
+        checkpoint = torch.load(checkpoint_path, map_location="cpu")
+        net_cpu.load_state_dict(checkpoint["state_dict"])
```

Comment:
Curious, I remember that in image classification checkpoints we save the name of the model architecture inside the checkpoint dict. Are we doing the same thing here? If yes, we could raise an error if the `args.basenet` deviates from the network loaded from the checkpoint. I imagine this could be quite good for the overall UX.

Reply:
Oh yeah great idea, I'll make that change